### PR TITLE
coprocessor_v2: FreeBSD uses .so

### LINKED ISSUE #17090: Building on FreeBSD fails
## Bug Report

```
error[E0425]: cannot find function `is_library_file` in this scope
   --> src/coprocessor_v2/plugin_registry.rs:133:24
    |
133 |                     if is_library_file(file) {
    |                        ^^^^^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find function `is_library_file` in this scope
   --> src/coprocessor_v2/plugin_registry.rs:248:20
    |
248 |                 if is_library_file(&file) {
    |                    ^^^^^^^^^^^^^^^ not found in this scope

For more information about this error, try `rustc --explain E0425`.
```

### What operating system and CPU are you using?

FreeBSD 14.0

### PR DESCRIPTION
**Pull request by dveeden**
### What is changed and how it works?

Issue Number: Close #17090 

What's Changed:

```commit-message

```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
Compatibility with FreeBSD has been restored for coprocessor_v2
```

### CODE REVIEW

File: src/coprocessor_v2/plugin_registry.rs [hunk 1]
```diff
@@ -463,6 +463,11 @@ fn is_library_file<P: AsRef<Path>>(path: P) -> bool {
     path.as_ref().extension() == Some(OsStr::new("so"))
 }
 
+#[cfg(target_os = "freebsd")]
```
Comment (BusyJay) [inline src/coprocessor_v2/plugin_registry.rs:466]: Better changing L461 to
```
#[cfg(any(target_os = "linux", target_os = "freebsd"))]
```

**Review by BusyJay — APPROVED** [review]

**Review by sticnarf — APPROVED** [review]

**Pull request merged**